anonlink/entitymatch.py

```diff
@@ -33,15 +33,26 @@ def python_filter_similarity(filters1, filters2):
 
 def cffi_filter_similarity_k(filters1, filters2, k, threshold):
     """Accelerated method for determining Bloom Filter similarity.
+
+    Assumes all filters are the same length, being a multiple of 64
+    bits.
+
     """
     length_f1 = len(filters1)
     length_f2 = len(filters2)
 
-    # We assume the length is 1024 bit = 128 Bytes
-    match_one_against_many_dice_1024_k_top = lib.match_one_against_many_dice_1024_k_top
+    if length_f1 == 0:
```

Comment:
Shouldn't we check `length_f2` as well?

Reply:
Hm, this might be a style thing. Normally I'd expect the function to be written in such a way that it produces the correct result when either list is empty, usually achieved easily because each statement ends up being a no-op when given an empty list; indeed such was true before this PR. The only reason I check that `filters1` is non-empty here is because I need to pick out its first element which in turn is just to obtain the filter length. Nothing in the rest of the code requires `len(filters2) != 0` so I would say that checking it is unnecessary.